docs/architecture/overview.md

```diff
@@ -0,0 +1,193 @@
+
+The [app model](../fundamentals/app-host-overview.md#define-the-app-model) serves as a blueprint for DCP to orchestrate your application. Under the hood, the app host is a .NET console application powered by the [📦 Aspire.Hosting.AppHost](https://www.nuget.org/packages/Aspire.Hosting.AppHost) NuGet package. This package includes build targets that register orchestration dependencies, enabling seamless dev-time orchestration.
+
+The .NET Aspire app host contains an implementation of the `k8s.KubernetesClient` (from the [📦 KubernetesClient](https://www.nuget.org/packages/KubernetesClient) NuGet package), which is a .NET client for Kubernetes. This client is used to communicate with the DCP API server, allowing the app host to delegate orchestration tasks to DCP.
+
+When you run the app host, DCP takes over, evaluates the app model, and orchestrates the resources. This process ensures that your application is fully configured and ready for local development, allowing you to focus on building features without worrying about infrastructure complexities. Consider the following diagram that helps to visualize the orchestration process:
+
+<span id="app-host-dcp-flow"></span>
+
+:::image type="content" source="media/app-host-dcp-flow-thumb.png" alt-text="A flow diagram depicting how the app host delegates to DCP." lightbox="media/app-host-dcp-flow.png":::
+
+For more information on the app host and APIs for building the app model, see [.NET Aspire orchestration overview](../fundamentals/app-host-overview.md).
+
+### Developer Control Plane
+
+DCP is at the core of the .NET Aspire app host orchestration functionality. It's responsible for orchestrating all resources defined in your app model, starting the developer dashboard, ensuring that everything is set up correctly for local development and testing. DCP manages the lifecycle of resources, applies network configurations, and resolves dependencies.
+
+DCP is written in Go, aligning with Kubernetes and its ecosystem, which are also Go-based. This choice enables deep, native integration with Kubernetes APIs, efficient concurrency, and access to mature tooling like Kubebuilder. DCP is delivered as two executables:
+
+- `dcp.exe`: API server that exposes a Kubernetes-like API endpoint for the app host to communicate with. Additionally, it exposes log streaming to the app host, which ultimately streams logs to the developer dashboard.
+- `dcpctrl.exe`: Controller that monitors the API server for new objects and changes, ensuring that the real-world environment matches the specified model.
+
+> [!NOTE]
+> DCP strives for "eventual consistency" rather than "strong consistency." This means that while DCP aims to keep the real-world environment in sync with the specified model, it might not always be perfectly aligned.
```

Comment:
It primarily means that changes to the model are applied asynchronously to the real world, and vice versa. There might be noticeable delay. DCP is trying very hard to match both worlds; it is not more lax than a "strongly consistent" system would be (and one might argue, it will do a better job since it will try multiple times before giving up).